src/vs/workbench/parts/terminal/electron-browser/windowsShellHelper.ts

```diff
@@ -79,16 +84,33 @@ export class WindowsShellHelper {
 	}
 
 	/**
-	 * Returns the innermost shell executable running in the terminal
+	 * Updates innermost shell executable and innermost shell running in the terminal
 	 */
-	public getShellName(): TPromise<string> {
+	public updateProgramName(): TPromise<void> {
 		if (this._childProcessIdStack.length === 0) {
 			this._childProcessIdStack.push(this._rootProcessId);
 		}
-		return new TPromise<string>((resolve) => {
-			this.refreshShellProcessTree(this._childProcessIdStack[this._childProcessIdStack.length - 1], null).then(result => {
-				resolve(result);
+		return new TPromise<void>((resolve) => {
+			this.refreshShellProcessTree(this._childProcessIdStack[this._childProcessIdStack.length - 1], this._shellName !== this._programName ? this._shellName : null).then(result => {
```

Comment:
Let's pull `this._shellName !== this._programName ? this._shellName : null` into a variable so it's easier to read